contracts/src/v0.8/workflow/dev/WorkflowRegistry.sol

```diff
@@ -0,0 +1,712 @@
+  /// - Updates the `secretsURL` hash mappings if the `secretsURL` changes.
+  ///
+  /// Requirements:
+  /// - `binaryURL` must always be provided, as it is required.
+  /// - If only one field is being updated, the other fields must be provided with their current values to keep them unchanged, otherwise
+  ///   they will be treated as empty strings.
+  /// - The DON ID must be in the allowed list to perform updates.
+  /// - The caller must be an authorized address. This means that even if the caller is the owner of the workflow, if they were later
+  ///   removed from the authorized addresses list, they will not be able to perform updates.
+  ///
+  /// Emits:
+  /// - `WorkflowUpdatedV1` event indicating the workflow has been successfully updated.
+  ///
+  /// @param workflowKey The unique identifier for the workflow.
+  /// @param newWorkflowID The rehashed unique identifier for the workflow.
+  /// @param binaryURL The URL pointing to the WASM binary. Must always be provided.
+  /// @param configURL The URL pointing to the configuration file. Provide an empty string ("") to remove it.
+  /// @param secretsURL The URL pointing to the secrets file. Provide an empty string ("") to remove it.
+  function updateWorkflow(
+    bytes32 workflowKey,
+    bytes32 newWorkflowID,
+    string calldata binaryURL,
+    string calldata configURL,
+    string calldata secretsURL
+  ) external registryNotLocked {
+    _validateWorkflowMetadata(newWorkflowID, bytes(binaryURL).length, bytes(configURL).length, bytes(secretsURL).length);
+
+    WorkflowMetadata storage workflow = _getWorkflowFromStorage(msg.sender, workflowKey);
+
+    uint32 donID = workflow.donID;
+    _validatePermissions(donID, msg.sender);
+
+    // Store the old workflowID for event emission.
+    bytes32 currentWorkflowID = workflow.workflowID;
+
+    // Condition to revert: WorkflowID must change, and at least one URL must change
+    if (currentWorkflowID == newWorkflowID) {
+      revert WorkflowIDNotUpdated();
+    }
+
+    // Determine which URLs have changed
+    bool sameBinaryURL = Strings.equal(workflow.binaryURL, binaryURL);
+    bool sameConfigURL = Strings.equal(workflow.configURL, configURL);
+    bool sameSecretsURL = Strings.equal(workflow.secretsURL, secretsURL);
+    if (sameBinaryURL && sameConfigURL && sameSecretsURL) {
+      revert WorkflowContentNotUpdated();
+    }
+
+    // Update all fields that have changed and the relevant sets
+    workflow.workflowID = newWorkflowID;
+    if (!sameBinaryURL) {
+      workflow.binaryURL = binaryURL;
+    }
+    if (!sameConfigURL) {
+      workflow.configURL = configURL;
+    }
+    if (!sameSecretsURL) {
+      // Remove the old secrets hash if secretsURL is not empty
+      if (bytes(workflow.secretsURL).length > 0) {
+        // Using keccak256 instead of _computeOwnerAndStringFieldHashKey as currentSecretsURL is memory
+        bytes32 oldSecretsHash = keccak256(abi.encodePacked(msg.sender, workflow.secretsURL));
```

Comment:
I didn't know encodePacked is being considered for depreciation, I guess it was chosen for a few reasons here
1) there's multiple inputs and this is more packed/ efficient than encode. 
2) encode is "safer" and better because the order matters, but in most of the cases in the registry contract it's not computed directly but instead with a helper function where the two params have different types so there's no way to mix them up and in that sense the order is already pre-determined.

But I guess to your point in this version tracker contract without the helper encode itself could totally work

Reply:
But interesting.... seems like for these specific use cases of hashing addresses with something else in particular encodePacked is the best usecase https://github.com/ethereum/solidity/issues/11593#issuecomment-1068070741 kind of a bummer about the removal?